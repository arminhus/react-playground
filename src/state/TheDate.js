import React from "react";

class TheDate extends React.Component {
  // The component is created.
  constructor(props) {
    super(props);
    this.state = { datetime: new Date() };
    console.log("constructor");
  }
  // The component has been added to the DOM
  componentDidMount() {
    console.log("componentDidMount");
    this.interval = setInterval(() => {
      this.setState({
        datetime: new Date()
      });
      // console.log("tick");
    }, 1000);
  }
  // The component is about to be removed from the DOM
  // When the parent components render doesn't render this
  // component anymore, the component will be removed.
  componentWillUnmount() {
    console.log("componentWillUnmount");
    clearInterval(this.interval);
  }
  // The component is reacting to being created or updated
  // 1 This will happen at least once.
  // 2 if the component props change, render can happen again.
  // 3  if the components state changes, the render can also happen again.
  render() {
    console.log("render");
    return <div>{this.state.datetime.toLocaleString()}</div>;
  }
}

export default TheDate;
