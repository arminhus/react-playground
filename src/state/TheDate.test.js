import React from "react";
import ReactDom from "react-dom";
import renderer from "react-test-renderer";
import TheDate from "./TheDate";

describe("TheDate Component Testing", () => {
  test("should render without crashing", () => {
    const div = document.createElement("div");
    ReactDom.render(<TheDate />, div);
    ReactDom.unmountComponentAtNode(div);
  });

  test.skip("should render the UI as expected from the snapshot", () => {
    expect(renderer.create(<TheDate />).toJSON()).toMatchSnapshot();
  });
});
