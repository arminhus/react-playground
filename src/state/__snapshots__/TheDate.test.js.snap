// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheDate Component Testing should render the UI as expected from the snapshot 1`] = `
<div>
  7/10/2019, 10:47:40 PM
</div>
`;
